build.gradle

```
@@ -2652,7 +2653,7 @@ project(':connect:runtime') {
   }
 
   task genConnectOpenAPIDocs(type: io.swagger.v3.plugins.gradle.tasks.ResolveTask, dependsOn: setVersionInOpenAPISpec) {
-    classpath = sourceSets.main.runtimeClasspath
+    classpath = sourceSets.main.compileClasspath
```

This feels like a terrible hack but by combining the runtime and compile classpaths, we can still generate the correct docs with `siteDocsTar` and omit the unnecessary Swagger dependencies from the release artifact generated with `releaseTarGz`:
```suggestion
    classpath = sourceSets.main.compileClasspath + sourceSets.main.runtimeClasspath

```

---

I am okay with this fix. Feel free to create a Jira to have a better fix beyond 3.3.0

---

The right fix may be to have a separate scope for swagger and that's basically runtimeClasspath + swagger deps.

---

I took a stab at that with https://github.com/ijuma/kafka/pull/7. This PR looks good enough to merge as-is but if you'd like to take a look, feel free.